### ISSUE TITLE
debug metadata for inlined functions

### ISSUE THREAD

**Issue by toivoh** [JuliaLang/julia#1189]
It seems that julia error tracebacks sometimes leave out some lines, including the one with the actual error.
When I load the following code from a script,

    module M
    function f()
        x        # line 3: error occurs here
    end
    function g()
        f()      # line 6
    end
    g()          # line 8
    end

it produces the output

    in g: x not defined
     in load at util.jl:230
     in load at util.jl:242
    at /work/toivo/code/juliastuff/code/errorline.jl:8
     in load at util.jl:253

Note that both line 3, where the error occurs, and line 6, which calls it, are missing from the traceback.

**Comment by mbauman** [comment]
I believe this can be closed since we now have the `--inline=no` option.

```julia
$ julia-0.4 --inline=no -q
julia> module M
       function f()
           x        # line 3: error occurs here
       end
       function g()
           f()      # line 6
       end
       g()          # line 8
       end
ERROR: UndefVarError: x not defined
 in f at none:3
 in g at none:6
 in eval at no file
```

**Comment by vtjnash** [comment]
Having a workaround isn't the same as fixing this

**Comment by mbauman** [comment]
Ah, right, sorry. As Jeff said:

> Yes I can imagine sticking in some extra metadata along with inlined function bodies. It's a bunch of tedious work but totally doable.

**Comment by ihnorton** [comment]
Ref #12544 -- if we change the inlining pass in inference to allow `:line` nodes through.

**Comment by ihnorton** [comment]
Fixed by @vtjnash in #13491 (with ~~rabble-rousing~~ moral support by me)
```
ERROR: UndefVarError: x not defined
 [inlined code] from none:3
 in g at none:6
```
